app/controllers/devices_controller.rb

```diff
@@ -20,7 +20,26 @@ def search
   # GET /devices/1
   # GET /devices/1.json
   def show
-    @device = Device.includes(:ips, :macs, :model, :brand, :plugin, :hive_queues, :related_devices => [ :ips, :macs, :model, :brand, :plugin, :hive_queues ]).find(params['id'])
+    respond_to do |format|
+      format.html do
+        @device = Device.includes(:ips, :macs, :model, :brand, :plugin, :hive_queues, :related_devices => [ :ips, :macs, :model, :brand, :plugin, :hive_queues ]).find(params['id'])
+      end
+      format.json do
+        if params[:view].present? && params[:view] == 'simple'
+          device = Device.includes(:model, :brand).find(params['id'])
+          @device = {
+            id: device.id,
+            name: device.name,
+            model: device.model ? device.model.name : nil,
+            brand: device.model && device.model.brand ? device.model.brand.name : nil,
```

Comment:
I think there's an association for brand through model, so you don't need to do the double-check
  brand: device.brand ? device.brand.name : nil,